source/common/http/conn_pool_grid.cc

```diff
@@ -334,5 +333,50 @@ void ConnectivityGrid::onDrainReceived() {
   }
 }
 
+bool ConnectivityGrid::shouldAttemptHttp3() {
+  if (http3_status_tracker_.isHttp3Broken()) {
+    ENVOY_LOG(trace, "HTTP/3 is broken to host '{}', skipping.", host_->hostname());
+    return false;
+  }
+  if (!alternate_protocols_.has_value()) {
+    // TODO(RyanTheOptimist): Make the alternate protocols cache required.
+    ENVOY_LOG(trace, "No alternate protocols cache. Attempting HTTP/3 to host '{}'.",
+              host_->hostname());
+    return true;
+  }
+  if (host_->address()->type() != Network::Address::Type::Ip) {
+    ASSERT(false);
+    return false;
+  }
+  uint32_t port = host_->address()->ip()->port();
+  // TODO(RyanTheOptimist): Figure out how scheme gets plumbed in here.
+  AlternateProtocolsCache::Origin origin("https", host_->hostname(), port);
+  OptRef<const std::vector<AlternateProtocolsCache::AlternateProtocol>> protocols =
+      alternate_protocols_->findAlternatives(origin);
+  if (!protocols.has_value()) {
+    ENVOY_LOG(trace, "HTTP/3 is not available to host '{}', skipping.", host_->hostname());
+    return false;
+  }
+
+  for (const AlternateProtocolsCache::AlternateProtocol& protocol : protocols.ref()) {
+    // TODO(RyanTheOptimist): Handle alternate protocols which change hostname or port.
+    if (!protocol.hostname_.empty() || protocol.port_ != port) {
+      continue;
```

Comment:
nit: add a trace log saying that we currently ignore these?

Reply:
Done.